gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -62,84 +70,65 @@ public static void main(String[] argv) throws IOException {
     } else {
-      log.error("Expected one argument, a parameter file.");
+      System.err.println(USAGE);
       System.exit(1);
     }
 
-    RestHighLevelClient client = null;
-
-    try {
-
-      client =
-          new RestHighLevelClient(
-              RestClient.builder(
-                  new HttpHost(
-                      parameters.getOptionalString(PARAM_HOSTNAME_PRIMARY).or(DEFAULT_HOST),
-                      parameters
-                          .getOptionalPositiveInteger(PARAM_PORT_PRIMARY)
-                          .or(DEFAULT_PORT_PRI),
-                      "http"),
-                  new HttpHost(
-                      parameters.getOptionalString(PARAM_HOSTNAME_SECONDARY).or(DEFAULT_HOST),
-                      parameters
-                          .getOptionalPositiveInteger(PARAM_PORT_SECONDARY)
-                          .or(DEFAULT_PORT_SEC),
-                      "http")));
-
-      String indexName = parameters.getString(PARAM_INDEX_NAME);
-
+    try (RestHighLevelClient client = buildElasticSearchClient(parameters)) {
+      final String indexName = parameters.getString(PARAM_INDEX_NAME);
       if (parameters.isPresent(PARAM_GIGAWORD_DIRECTORY_PATH)) {
-        File gigawordDir = parameters.getExistingDirectory(PARAM_GIGAWORD_DIRECTORY_PATH);
-        for (File concatenatedFileToIndex : gigawordDir.listFiles()) {
-          // Non Gzip files in the directory are ignored
-          if (concatenatedFileToIndex.isFile() && isValidGzipFile(concatenatedFileToIndex)) {
-            index(client, concatenatedFileToIndex, indexName);
-          }
+        final PathMatcher gzippedConcatenatedFilePattern = FileSystems.getDefault().getPathMatcher(
+                "glob:**/data/**/*.gz");
+        final Path gigawordDir = parameters.getExistingDirectory(PARAM_GIGAWORD_DIRECTORY_PATH).toPath();
+        try (Stream<Path> gigawordFiles = Files.walk(gigawordDir)) {
+          gigawordFiles
+                  .filter(gzippedConcatenatedFilePattern::matches)
+                  .forEach(gzippedConcatenatedFile -> {
+                    try {
+                          index(client, gzippedConcatenatedFile, indexName);
+                    } catch (IOException e) {
+                      throw new RuntimeException(e);
+                    }
+                  });
         }
       } else {
         File concatenatedFileToIndex = parameters.getExistingFile(PARAM_GIGAWORD_FILEPATH);
-        if (!isValidGzipFile(concatenatedFileToIndex)) {
-          throw new IOException(
-              concatenatedFileToIndex.getAbsolutePath() + " is not a valid GZip file");
-        }
-        index(client, concatenatedFileToIndex, indexName);
+        index(client, concatenatedFileToIndex.toPath(), indexName);
       }
 
     } catch (Exception e) {
       log.error("Caught exception: {}", e);
       System.exit(1);
-    } finally {
-      if (client != null) {
-        client.close();
```

Comment:
Was it intentional to leave `client.close()` out?

Reply:
It gets called automatically by the [try-with-resources ](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) statement.